src/libraries/Microsoft.PowerFx.Interpreter/Functions/ResolvedObjectHelpers.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Core.Public;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+using Microsoft.PowerFx.Core.Types;
+
+namespace Microsoft.PowerFx.Functions
+{
+    // Core operations for turning ResolvedObjects into PowerFx values
+    internal static class ResolvedObjectHelpers
+    {
+        public static FormulaValue RecalcFormulaInfo(RecalcFormulaInfo fi)
+        {
+            return fi._value;
+        }
+
+        public static FormulaValue OptionSet(OptionSet optionSet, IRContext irContext)
+        {
+            var options = new List<NamedValue>();
+            foreach (var option in optionSet.Options)
+            {
+                if (!optionSet.TryGetValue(option.Key, out var osValue))
+                {
+                    return CommonErrors.UnreachableCodeError(irContext); 
```

Comment:
Can't connect "unreachable code" with failure to resolve value of an option. Comment might help